### PR TITLE
Update docs for v0.5.3

### PR DESCRIPTION
**Pull request by adrianludwin**
Update all references to v0.5.3 and document the user-facing changes.

Staged at:
* https://github.com/adrianludwin/multi-tenancy/blob/docs-053/incubator/hnc/README.md
* https://github.com/adrianludwin/multi-tenancy/blob/docs-053/incubator/hnc/docs/user-guide/how-to.md

### CODE REVIEW

File: incubator/hnc/docs/user-guide/how-to.md [hunk 1]
```diff
@@ -509,6 +508,10 @@ HNC propagates `Roles` and `RoleBindings` by default. You can also set any type
 of Kubernetes resource to any of the propagation modes discussed above. To do
 so, you need cluster privileges.
 
+**WARNING: If you start propagating a new object type, HNC _cannot_ check
+whether there are conflicting objects in descendant namespaces, and will
+overwrite them. This will be fixed in HNC v0.6 (see #1076).**
```
Comment (yiqigao217) [inline incubator/hnc/docs/user-guide/how-to.md:513]: #1102 may be more specific to this warning, but your call to reference the object overwriting issue (#1076) or the hncconfig update mode issue (#1103) here.

**Review by yiqigao217 — COMMENTED** [review]
/lgtm
/hold

Just left a minor discussion. @rjbez17 , @adrianludwin , feel free to remove the hold.

File: incubator/hnc/docs/user-guide/how-to.md (hunk 1 above)
Comment (adrianludwin) [reply incubator/hnc/docs/user-guide/how-to.md]: Done

**Pull request merged**